HealthKitReporter/Classes/Model/HKUnitConvertable.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  HKUnitConvertable.swift
+//  
+//
+//  Created by Vignesh J on 16/02/21.
+//
+
+import Foundation
+import HealthKit
+
+public protocol HKUnitConvertable {
+    func converted(to unit: HKUnit) throws -> Self
```

Comment:
Also I would use instead of HKUnit type, the String, because the idea of this pod is too be suitable as a resource pod for the flutter plugin https://github.com/VictorKachalov/health_kit_reporter

The pluging will require to have a bridging function which will accept the String parameter in order to use it on flutter side and will pass the string as well if the Quantity object is created on a flutter side and needs to be converted to another unit. So the HKUnit object creation can be incapsulated inside the "extension Quantity: UnitConvertable", so there will be no need to even import original HealthKit , only this pod